### PR TITLE
Include data for new 'Studienordnung 23'

### PR DESCRIPTION
**Pull request by jeremystucki**
Modules are kept global as we don't want to deal with migrating those. Only categories and focuses are specific to each Studienordnung.

I recommend looking at the individual commits to make the review process easier.

### CODE REVIEW

File: crawler.py [hunk 1]
```diff
@@ -3,126 +3,146 @@
...
-        for cat in module['categories']:
-            categories[cat['id']]['modules'].append({'id': module['id'], 'name': module['name'],'url': module['url']})
-            categories[cat['id']]['total_ects'] += module['ects']
+        if 'kategorien' in zuordnung:
+            module['categories'] = [
+                {'id': getIdForCategory(z['kuerzel']), 'name': z['bezeichnung'], 'ects': z['kreditpunkte']} for z in
+                zuordnung['kategorien']]
+            module['ects'] = zuordnung['kategorien'][0]['kreditpunkte']
+
+        if module['id'].endswith('_p'):
```
Comment (StefanieJaeger) [inline crawler.py:58]: Add comment

File: crawler.py [hunk 2]
```diff
@@ -3,126 +3,146 @@
...
+
+    # load more infos about modules
+    for module in modules.values():
+        moduleContent = json.loads(requests.get(f'{BASE_URL}{module["url"]}').content)
+
+        # needed for modules, whose credits do not count towards "Studiengang Informatik"
+        if 'kreditpunkte' in moduleContent and module['ects'] == 0:
+            module['ects'] = moduleContent['kreditpunkte'];
+
+        if module['id'].startswith('Kat'):
```
Comment (StefanieJaeger) [inline crawler.py:71]: Add comment

File: data21/focuses.json [hunk 3]
```diff
@@ -43,6 +43,11 @@
         "id": "SecSoW",
         "name": "Secure Software",
         "url": "allModules/40311_M_SecSoW.json"
+      },
+      {
+        "id": "SecSW",
```
Comment (jeremystucki) [inline data21/focuses.json:48]: Reminder for myself: Get rid of one of the SecSW modules

File: crawler.py (hunk 1 above)
Comment (jeremystucki) [inline crawler.py:58]:
```suggestion
        # These are the new IKTS modules. They are split into two separate modules, one of them being a "Projektarbeit".
        # This ensures that they can be differentiated in the UI.
        if module['id'].endswith('_p'):
```

Comment (jeremystucki) [reply crawler.py:58]: @StefanieJaeger agree with the comment?

File: crawler.py (hunk 2 above)
Comment (jeremystucki) [inline crawler.py:71]:
```suggestion
        # For some reason each category is also present as a module.
        # This filters them out.
        if module['id'].startswith('Kat'):
```

Comment (jeremystucki) [reply crawler.py:71]: @StefanieJaeger agree with the comment?

**Review by StefanieJaeger — APPROVED** [review]

**Pull request merged**